uaibank.c

```diff
@@ -29,52 +29,122 @@ User new_user(int users_qnty) {
   user.age = age;
   user.current_balance = current_balance;
 
-  printf("\nUsuário 0%d adicionado com sucesso\n\n", user.id);
+  printf("\nUsuário %d adicionado com sucesso\n\n", user.id);
 
   return user;
 }
 
-void main() {
+void search_id(int id, User users[], int users_qnty) {
+  int found = 0; 
+
+  for (int i = 0; i < users_qnty; i++) {
+    if (users[i].id == id) {
+      printf("ID: %d\nNome: %s\nIdade: %d\nSaldo atual: %.2f\n", users[i].id, users[i].name, users[i].age, users[i].current_balance);
+      found = 1;
+      break; 
+    }
+  }
+
+  if (!found) {
+    printf("\nErro: Usuário %d não encontrado.\n", id);
+  }
+}
+
+void transfer_money(int id_origin, int id_destination, User users[], int users_qnty, float amount) {
+  for (int i = 0; i < users_qnty; i++) {
+    if (users[i].id == id_origin) {
+      users[i].current_balance -= amount;
+    }
+    if (users[i].id == id_destination) {
+      users[i].current_balance += amount;
+    }
+  }
+  //é pra printar os usuário que fizeram a troca?
+  printf("Transferência entre usuários feita com sucesso!\n");
+}
+
```

Comment:
aqui da pra usar a funcao `search_id` pra pegar o usuario de origem e de destino

Reply:
to fazendo, assim que acabar eu subo aqui

Reply:
pegar o search_id pra isso vai levar um trampo por alguns motivos. 1- a função search_id não retorna o usuário no final, ele printa no meio da função, 2- pra fazer o search retornar o usuário ele teria que retornar um vetor, e como o nome e string, idade int e saldo float, um vetor normal não ia dar conta, a gente meio que teria que criar uma struct dentro do search e retornar o indice 0 dela. Ai poderia usar no tranfer_money, mas tipo isso séria um trampinho, ( no quesito ter que gastar mais memória) mas se bem que fazer vários for não é tão bom no quesito computacional, enfim, esse é minha visão, se vc tiver outra ideria de como fazer isso podemos conversar hj no lab as 16, (se fosse em pyhton n ia precisar td isso :(